analyzer/src/namespace/types.rs

```diff
@@ -182,37 +182,48 @@ impl Struct {
         Struct {
             name: name.to_string(),
             fields: BTreeMap::new(),
+            order: vec![],
```

Comment:
I'm surprised this is needed. The reason I chose `BTreeMap` was because I thought it is ordered. I guess if that doesn't hold we could also change it back to a regular `HashMap`.

Reply:
They are ordered, as in, they implement `Ord`. However, this only means that the map objects themselves can be compared and sorted. Regular hashmaps don't implement `Ord` and therefore can't be compared or sorted. Being able to sort types is actually required at the moment for dispatching ABI runtime functions, since we want to remove any duplicates. Removal of duplicates is accomplished by first sorting the a collection of parameter types and the de-duplicating the collection.